pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -245,7 +245,7 @@ public void reloadSegment(String tableNameWithType, String segmentName, boolean
   }
 
   @Override
-  public void reloadAllSegments(String tableNameWithType, boolean forceDownload,
+  public void reloadAllSegments(String taskId, String tableNameWithType, boolean forceDownload,
```

Comment:
Let's also add `taskId` to the single `reloadSegment()` method

Reply:
Ack. I'm planning to add it post getting a general agreement on the approach with all segments APIs first. Is that okay?